app/models/miq_queue.rb

```diff
@@ -332,7 +332,7 @@ def self.put_or_update(find_options)
                     args = YAML.dump(conds.delete(:args))
                     MiqQueue.where(conds).where(['args = ?', args])
                   else
-                    MiqQueue.where(conds)
+                    MiqQueue.where(conds).select(MiqQueue.columns.map(&:name) - ['jsonb_data'])
```

Comment:
Why here but not in the other conditional?

Reply:
Focusing on refresh path, not sure where is the other condition used, but it will fetch blank jsonb_data, which is ok. This is for avoiding fetching all of the targets(can be thousands) everytime we add 1 new.

Reply:
> MiqQueue.columns.map(&:name)

Prefer `MiqQueue.column_names`  (FYI, if you need symbols, `MiqQueue.column_names_symbols`)

Reply:
You probably don't want to calculate that list in a loop either, so you can make that a memoized class method somewhere (I'd would have said a constant, but we can access the database on model load, so you want to delay it with a memoized class method).

Reply:
fixed in the other PR